### PR TITLE
✅🐛 Poll for suspended flow nodes in the start event tests

### PR DESCRIPTION
**Pull request by robpal94**
**Changes:**

1. Poll for suspended FlowNodeInstances before running the tests for MessageStartEvents and SignalStartEvents. 
This should help avoid running into the timeout issue that sometimes causes Jenkins builds to fail.

**Issues:**

Closes: #123

PR: #127

## How can others test the changes?

Run the tests.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: test/1_process_engine/start_event_tests.js [hunk 1]
```diff
@@ -121,6 +121,31 @@ describe('Start Events - ', () => {
...
+    const retryDelay = 500;
+
+    const flowNodeInstanceService = await testFixtureProvider.resolveAsync('FlowNodeInstanceService');
+
+    for (let currentTry = 0; currentTry < maxNumberOfRetries; currentTry += 1) {
+      await wait(retryDelay);
+
+      let flowNodeInstances = await flowNodeInstanceService.querySuspendedByCorrelation(correlationId);
+
+      if (processModelId) {
```
Comment (Vyperus) [inline test/1_process_engine/start_event_tests.js]: Extract business rule.

File: test/1_process_engine/start_event_tests.js [hunk 2]
```diff
@@ -121,6 +121,31 @@ describe('Start Events - ', () => {
...
+        flowNodeInstances = flowNodeInstances.filter((fni) => {
+          return fni.tokens[0].processModelId === processModelId;
+        });
+      }
+
+      if (flowNodeInstances.length >= 1) {
```
Comment (Vyperus) [inline test/1_process_engine/start_event_tests.js]: Extract business rule.

File: test/1_process_engine/start_event_tests.js [hunk 3]
```diff
@@ -121,6 +121,31 @@ describe('Start Events - ', () => {
...
+        break;
+      } else {
+        throw new Error(`No process instance within correlation '${correlationId}' found! The process instance may failed to start!`);
```
Comment (Vyperus) [inline test/1_process_engine/start_event_tests.js]: Is a generic error correct here? @ElRaptorus

File: test/1_process_engine/start_event_tests.js [hunk 4]
```diff
@@ -121,6 +121,31 @@ describe('Start Events - ', () => {
...
+    const flowNodeInstanceService = await testFixtureProvider.resolveAsync('FlowNodeInstanceService');
+
+    for (let currentTry = 0; currentTry < maxNumberOfRetries; currentTry += 1) {
+      await wait(retryDelay);
+
+      let flowNodeInstances = await flowNodeInstanceService.querySuspendedByCorrelation(correlationId);
+
+      if (processModelId) {
+        flowNodeInstances = flowNodeInstances.filter((fni) => {
+          return fni.tokens[0].processModelId === processModelId;
```
Comment (Vyperus) [inline test/1_process_engine/start_event_tests.js]: Extract business rule.

File: test/1_process_engine/start_event_tests.js (hunk 3 above)
Comment (ElRaptorus) [reply test/1_process_engine/start_event_tests.js]: This error is only thrown within the test suite and has nothing to do with the ProcessEngine itself.
There really isn't much point in annotating it with error codes, imho.

**Pull request merged**